Cargo.toml

```diff
@@ -27,6 +27,7 @@ thiserror = "1"
 smartstring = { version = "1" }
 serde_json = { version = "1" }
 either = "1.9"
+deltalake = "0.17.1"
```

Comment:
I'm a bit hesitant to add this in as a dependency. 

this pulls in the entire arrow-rs project as well as many other heavy dependencies.

Reply:
It makes sense, I will keep looking for alternatives. Thx

Reply:
FWIW, I think the "ideal" approach would be first to upstream some work to delta-rs and expose a javascript library. Then we could use the JS library directly like py-polars does, and do a bunch of the logic on the JS side using `apache-arrow` as an optional dependency. 


If the `apache-arrow` npm package becomes a point of contention, we could likely bypass it by dynamically linking to the delta js bindings directly.

Reply:
`delta-rs` team is busy with C# implementation now. :)

Reply:
There was effort to use [delta-sharing](https://github.com/delta-incubator/delta-sharing-connectors) using JS and [Danfo.js](https://danfo.jsdata.org/).